os/arch/arm/src/armv7-m/up_memfault.c

```diff
@@ -109,36 +116,36 @@ int up_memfault(int irq, FAR void *context, FAR void *arg)
 	/* Dump some memory management fault info */
 
 	(void)irqsave();
-	lldbg("PANIC!!! Memory Management Fault:\n");
-	mfdbg("  IRQ: %d context: %p\n", irq, regs);
-	lldbg("  CFAULTS: %08x MMFAR: %08x\n",
-		  getreg32(NVIC_CFAULTS), getreg32(NVIC_MEMMANAGE_ADDR));
-	mfdbg("  BASEPRI: %08x PRIMASK: %08x IPSR: %08x CONTROL: %08x\n",
-		  getbasepri(), getprimask(), getipsr(), getcontrol());
-	mfdbg("  R0: %08x %08x %08x %08x %08x %08x %08x %08x\n",
-		  regs[REG_R0], regs[REG_R1], regs[REG_R2], regs[REG_R3],
-		  regs[REG_R4], regs[REG_R5], regs[REG_R6], regs[REG_R7]);
-	mfdbg("  R8: %08x %08x %08x %08x %08x %08x %08x %08x\n",
-		  regs[REG_R8], regs[REG_R9], regs[REG_R10], regs[REG_R11],
-		  regs[REG_R12], regs[REG_R13], regs[REG_R14], regs[REG_R15]);
-
-#ifdef CONFIG_ARMV7M_USEBASEPRI
-#ifdef REG_EXC_RETURN
-	mfdbg("  xPSR: %08x BASEPRI: %08x EXC_RETURN: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_BASEPRI], current_regs[REG_EXC_RETURN]);
-#else
-	mfdbg("  xPSR: %08x BASEPRI: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_BASEPRI]);
-#endif
-#else
-#ifdef REG_EXC_RETURN
-	mfdbg("  xPSR: %08x PRIMASK: %08x EXC_RETURN: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_PRIMASK], current_regs[REG_EXC_RETURN]);
-#else
-	mfdbg("  xPSR: %08x PRIMASK: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_PRIMASK]);
-#endif
-#endif
+	uint32_t *regs = (uint32_t *)context;
+
+	uint32_t cfsr = getreg32(NVIC_CFAULTS);
+	uint32_t mmfar = getreg32(NVIC_MEMMANAGE_ADDR);
+
+	lldbg("PANIC!!! Memory Management Fault occured while executing instruction at address : 0x%08x\n", regs[REG_R15]);
+	lldbg("CFAULTS: 0x%08x MMFAR: 0x%08x\n", cfsr, mmfar);
+
+	if(cfsr & MASK_DACCVIOL){
+		lldbg("Data access violation occured.\n");
+	}
+	else if(cfsr & MASK_IACCVIOL){
+		lldbg("Instruction access violation occured while fetching instruction from an Execute Never (XN) region.\n");
+	}
+	else if(cfsr & MASK_MSTKERR){
+		lldbg("Memfault occurred while stacking registers during exception entry.\n");
+	}
+	else if(cfsr & MASK_MUNSTKERR){
+		lldbg("Memfault occurred while unstacking registers during exception return.\n");
+	}
+	else if(cfsr & MASK_MLSPERR){
+		lldbg("Memfault occurred during lazy state preservation of Floating Point unit registers.\n");
+	}
+
+	if(cfsr & MASK_MMARVALID){
+		lldbg("Access violation occured at address : 0x%08x\n", mmfar);
```

Comment:
`mmfar` is printed in line 125.
Can we merge these lines with it?

Reply:
Done.